packages/build-tools/src/common/installDependencies.ts

```diff
@@ -47,6 +47,10 @@ export async function installDependenciesAsync({
     default:
       throw new Error(`Unsupported package manager: ${packageManager}`);
   }
+  if (env['EAS_YARN_FOCUS_WORKSPACE']) {
+    // override package manager to yarn if this flag is set
+    [packageManager, ...args] = env['EAS_YARN_FOCUS_WORKSPACE'].split(' ') as [PackageManager, ...string[]];
+  }
```

Comment:
how about sth like


```suggestion
  if (env['EAS_YARN_FOCUS_WORKSPACE']) {
    if (packageManager != yarn) {
      logger.warn("EAS_YARN_FOCUS_WORKSPACE is set, but the workspace is not using yarn.")
    } else {
      logger.info(`EAS_YARN_FOCUS_WORKSPACE is set to '${...}', changing install command to ...`);
      args = ["workspaces", "focus", env['EAS_YARN_FOCUS_WORKSPACE']];
    }
  }
```
?